File: data/src/main/java/com/example/data/api/AuthApi.java
package com.example.data.api;

import com.example.data.dto.LoginRequest;
import com.example.data.dto.SignUpRequest;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface AuthApi {

    /**
     * 회원가입
     *
     * @param signUpRequest 회원가입 요청 정보
     * @return String 회원 JWT 토큰
     */
    @POST("/api/auth/signup")
    Call<String> signup(
            @Body SignUpRequest signUpRequest
    );

    /**
     * 로그인
     *
     * @param loginRequest 로그인 요청 정보
     * @return String 회원 JWT 토큰
     */
    @POST("/api/auth/login")
    Call<String> login(
            @Body LoginRequest loginRequest
    );

    @POST("/api/auth/findEmailByPhone")
    Call<String> findEmailByPhone(
            @Query("phone") String phone
    );

    @POST("/api/auth/email-check")
    Call<Boolean> emailCheck(
            @Query("email") String email
    );

}

File: data/src/main/java/com/example/data/api/AuthApi.kt
package com.example.data.api

import com.example.data.dto.EmailResponse
import com.example.data.dto.IdCheckResponse
import com.example.data.dto.LoginRequest
import com.example.data.dto.SignUpRequest
import com.example.data.dto.TokenResponse
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Query

interface AuthApi {

    @POST("/api/auth/login")
    suspend fun login(
        @Body loginRequest: LoginRequest
    ): Response<TokenResponse>

    @POST("/api/auth/signup")
    suspend fun signUp(
        @Body signUpRequest: SignUpRequest
    ): Response<TokenResponse>

    @POST("/api/auth/findEmailByPhone")
    suspend fun findEmailByPhone(
        @Query("phone") phone: String
    ): Response<EmailResponse>

    @POST("/api/auth/email-check")
    suspend fun emailDuplicateCheck(
        @Query("email") email: String
    ): Response<IdCheckResponse>
}